google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfiguration.java

```diff
@@ -235,12 +237,18 @@ public int hashCode() {
   }
 
   private void checkCommonConfig(AppEngineDeployable deployable) throws RuntimeConfigurationError {
-    Set<CloudSdkValidationResult> sdkValidationResult =
-        CloudSdkValidator.getInstance().validateCloudSdk();
-    if (!sdkValidationResult.isEmpty()) {
-      CloudSdkValidationResult result = Iterables.getFirst(sdkValidationResult, null);
-      throw new RuntimeConfigurationError(
-          GctBundle.message("appengine.flex.config.server.error", result.getMessage()));
+    // do not check SDK if it supports dynamic install - the deployment runner will block itself
+    // until installation is done.
+    CloudSdkService cloudSdkService = CloudSdkService.getInstance();
+    SdkStatus sdkStatus = cloudSdkService.getStatus();
+    if (sdkStatus != SdkStatus.READY && !cloudSdkService.isInstallReady()) {
```

Comment:
so you only check for an invalid SDK if the sdk is not ready and the service is no ready to install? could you explain this? Im having a hard time understanding.

Reply:
The idea is deployment configuration should be marked invalid only if SDK is invalid (not ready) and cannot re-install itself to fix it (i.e. custom SDK path). In this case the configuration is blocked and a user is asked to fix it himself. Otherwise we'll try to fix it by re-installing the SDK. Hopefully this makes it a bit clearer? Open to suggestions how to handle this validation case cleaner.

Reply:
what if the install is in progress? Will `CloudSdkService#isInstallReady` not return false in this case (and thus allowing the config here to mark itself invalid)?

Reply:
`isInstallReady` will be true in this case, it means 'ready to install in general', not at the specific moment of call, so the check will not pass.

Reply:
in that case I think we should rename `isInstallReady`. Do `isInstallable` or `isInstallAvailable` capture the meaning you are after?

Reply:
i think `isInstallable` is best, assuming it captures the correct meaning

Reply:
I think it's better. I remember we started with `supportsInstall` but then decided to change to `isInstallReady` so the truth is somewhere in there. `isInstallable` sounds correct.